source-mongodb/pull.go

```diff
@@ -734,37 +734,56 @@ func eventResourceId(ev changeEvent) string {
 const maxTimeMilli = 253402300799999
 const minTimeMilli = -62167219200000
 
+func sanitizePrimitive(input interface{}) interface{} {
+	switch v := input.(type) {
+	case primitive.DateTime:
+		if v < minTimeMilli {
+			return primitive.DateTime(minTimeMilli)
+		} else if v > maxTimeMilli {
+			return primitive.DateTime(maxTimeMilli)
+		}
+	case float64:
+		if math.IsNaN(v) {
+			return "NaN"
+		} else if math.IsInf(v, +1) {
+			return "Infinity"
+		} else if math.IsInf(v, -1) {
+			return "-Infinity"
+		}
+	case map[string]interface{}:
+		return sanitizeDocument(v)
```

Comment:
Could we recursively call `sanitizePrimitive` here without going back around through `sanitizeDocument`?